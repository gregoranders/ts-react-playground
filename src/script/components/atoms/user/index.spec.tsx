import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';

import { DefaultUser } from '@models/user';

import * as TestSubject from './index';

describe(`${TestSubject.UserView.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  const OnClick = jest.fn();

  afterEach(() => {
    OnClick.mockReset();
  });

  it('snapshot', (): void => {
    const snapshot = create(<TestComponent user={DefaultUser} />);
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('mount', () => {
    const testSubject = mount(<TestComponent user={DefaultUser} />);
    expect(testSubject.text()).toBe('First NameLast NameeMail');
  });
});
