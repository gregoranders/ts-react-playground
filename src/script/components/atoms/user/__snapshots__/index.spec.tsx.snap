// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView snapshot 1`] = `
<dl>
  <dt>
    First Name
  </dt>
  <dd>
    
  </dd>
  <dt>
    Last Name
  </dt>
  <dd>
    
  </dd>
  <dt>
    eMail
  </dt>
  <dd>
    
  </dd>
</dl>
`;
